packages/hebao_v2/contracts/thirdparty/proxy/WalletProxy.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Taken from: https://github.com/gnosis/safe-contracts/blob/development/contracts/proxies/GnosisSafeProxy.sol
+pragma solidity ^0.7.0;
+
+/// @title IProxy - Helper interface to access masterCopy of the Proxy on-chain
+/// @author Richard Meissner - <richard@gnosis.io>
+interface IProxy {
```

Comment:
This interface is not used anywhere...

Reply:
It's a trick for the proxy. `address internal masterCopy;` is made internal, because making it public would auto generate getter methods that would increase the proxy contract size, so deployment costs increase (I tested this out, is makes a significant difference, I think it saves 2 storage slots). But `masterCopy` is still public because the function is implemented in the `fallback` methdod:
```
// 0xa619486e == keccak("masterCopy()"). The value is right padded to 32-bytes with 0s
if eq(calldataload(0), 0xa619486e00000000000000000000000000000000000000000000000000000000) {
   mstore(0, _masterCopy)
   return(0, 0x20)
}
```

So the proxy contract doesn't implement the interface from solidity's point of view, but it does actually implement it in a non-standard way so the interface can be used to interact with the proxy contract.